tests/contrib/test_pydantic.py

```diff
@@ -0,0 +1,44 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from pydantic import BaseModel, Field
+from typing_extensions import Annotated
+
+from litestar.contrib.pydantic import PydanticDTO
+from litestar.dto.factory import dto_field
+from litestar.dto.factory.types import FieldDefinition
+from litestar.typing import ParsedType
+
+if TYPE_CHECKING:
+    from typing import Callable
+
+
+def test_field_definition_generation(
+    int_factory: Callable[[], int], expected_field_defs: list[FieldDefinition]
+) -> None:
+    class TestModel(BaseModel):
+        a: int
+        b: Annotated[int, Field(**dto_field("read-only"))]
+        c: Annotated[int, Field(gt=1)]
+        d: int = Field(default=1)
+        e: int = Field(default_factory=int_factory)
+
+    field_defs = list(PydanticDTO.generate_field_definitions(TestModel))
+    assert (
+        field_defs[0].unique_model_name
+        == "tests.contrib.test_pydantic.test_field_definition_generation.<locals>.TestModel"
+    )
+    for field_def, exp in zip(field_defs, expected_field_defs):
+        assert field_def == exp
+
```

Comment:
So, we also need tests for the openapi docs imo.

Reply:
I've not done explicit openapi output tests as I'm using the existing `_openapi.schema_generation.schema.create_schema()` function . If there is some DTO specific stuff introduced throughout [this](https://github.com/litestar-org/litestar/pull/1712#discussion_r1201294174) I'll ensure it is tested.

How would you feel about exposing a `create_schema()` function outside of the private `_open_api` module, it can simply call into the private one, but would allow that api to be solidified.